### PR TITLE
Deprecate the galgebra.deprecated module

### PR DESCRIPTION
**Pull request by eric-wieser**

### CODE REVIEW

File: examples/ipython/Old Format.ipynb [hunk 1]
```diff
@@ -15,7 +15,12 @@
     "\n",
     "def check_latex(name):\n",
     "    # todo: use subprocess.run once we drop Python 2.7\n",
-    "    p = subprocess.Popen([sys.executable, name + '.py'], stderr=subprocess.PIPE, universal_newlines=True)\n",
+    "    p = subprocess.Popen([\n",
+    "        sys.executable,\n",
+    "        # warning filter, because we expect deprecations when testing the old API \n",
+    "        \"-Wignore:galgebra.deprecated:DeprecationWarning\",\n",
```
Comment (eric-wieser) [inline examples/ipython/Old Format.ipynb:21]:
```suggestion
    "        r\"-Wignore:.*galgebra\\.deprecated:DeprecationWarning\",\n",
```

**Comment by utensil** [comment]
Before we actually deprecate them, we need to ensure we have equivalent tests outside `Old Format`, it provided a significant coverage boost.

The approach could be:

1. Start by moving theses classes out of GAlgebra to test as merely a test facility
2. Remove the duplicated tests in `Old Format`;
3. Port the tests the increases coverage to the new format.

**Comment by eric-wieser** [comment]
In my opinion there's no need to do those things before _deprecating_`galgebra.deprecated`, but you're right that we should do then before removing it entirely.

**Comment by utensil** [comment]
Yes, I only meant before removing them not deprecating them.

**Comment by eric-wieser** [comment]
Blocked by gh-347

**Pull request merged**